persistence-cassandra/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/cassandra/CassandraReadSideHandler.scala

```diff
@@ -28,44 +29,57 @@ import scala.concurrent.{ ExecutionContext, Future }
  * Internal API
  */
 private[cassandra] abstract class CassandraReadSideHandler[Event <: AggregateEvent[Event], Handler](
-  session: CassandraSession, handlers: Map[Class[_ <: Event], Handler], dispatcher: String
+  session:    CassandraSession,
+  handlers:   Map[Class[_ <: Event], Handler],
+  dispatcher: String
 )(implicit ec: ExecutionContext) extends ReadSideHandler[Event] {
 
   private val log = LoggerFactory.getLogger(this.getClass)
 
   protected def invoke(handler: Handler, event: Event, offset: Offset): CompletionStage[JList[BoundStatement]]
 
   override def handle(): Flow[Pair[Event, Offset], Done, _] = {
-    akka.stream.scaladsl.Flow[Pair[Event, Offset]].mapAsync(parallelism = 1) { pair =>
-      handlers.get(pair.first.getClass.asInstanceOf[Class[Event]]) match {
-        case Some(handler) =>
-          for {
-            statements <- invoke(handler, pair.first, pair.second).toScala
-            done <- statements.size match {
-              case 0 => Future.successful(Done.getInstance())
-              case 1 => session.executeWrite(statements.get(0)).toScala
-              case _ =>
-                val batch = new BatchStatement
-                val iter = statements.iterator()
-                while (iter.hasNext)
-                  batch.add(iter.next)
-                session.executeWriteBatch(batch).toScala
+
+    def executeStatements(statements: JList[BoundStatement]): Future[Done] = {
+      val batch = new BatchStatement
+      // statements is never empty, there is at least the store offset statement
```

Comment:
I couldn't out if this could be a problem, so added a if/else check. 

This only affect the javadsl because the scala one does not have a legacy version